File: collabwrk-backend/app/api/company.py
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy.ext.asyncio import AsyncSession
from app.database import get_db
from app.schemas.company import CompanyCreate, CompanyResponse, InviteCode, CompanySeatsResponse
from app.services.company_service import create_company, join_company, get_company_seats
from app.services.auth_service import get_current_user

router = APIRouter()

@router.post("/create", response_model=CompanyResponse)
async def create_new_company(
    company_in: CompanyCreate,
    db: AsyncSession = Depends(get_db),
    current_user_and_token = Depends(get_current_user)
):
    user, _ = current_user_and_token
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
        
    company = await create_company(
        db, 
        name=company_in.name, 
        creator_id=user.id,
        max_seats=company_in.max_seats
    )
    return company

@router.post("/join", response_model=CompanyResponse)
async def join_existing_company(
    invite: InviteCode,
    db: AsyncSession = Depends(get_db),
    current_user_and_token = Depends(get_current_user)
):
    user, _ = current_user_and_token
    if not user:
        raise HTTPException(status_code=401, detail="Not authenticated")
        
    company = await join_company(db, user_id=user.id, invite_code=invite.invite_code)
    return company

@router.get("/seats", response_model=CompanySeatsResponse)
async def read_company_seats(
    db: AsyncSession = Depends(get_db),
    current_user_and_token = Depends(get_current_user)
):
    user, _ = current_user_and_token
    if not user or not user.company_id:
        raise HTTPException(status_code=400, detail="User does not belong to a company")
        
    seats_data = await get_company_seats(db, company_id=user.company_id)
    return seats_data

File: collabwrk-native/app/services/api/company.ts
import apiClient from './client';

export interface Company {
    id: string;
    name: string;
    invite_code: string;
    max_seats: number;
}

export interface Seat {
    id: string;
    company_id: string;
    user_id?: string;
    status: string;
    assigned_at?: string;
    user?: {
        first_name: string;
        last_name: string;
        email: string;
    }
}

export interface CompanySeatsResponse {
    seats: Seat[];
    available_count: number;
    total_count: number;
}

export const companyApi = {
    create: async (name: string, maxSeats: number = 10): Promise<Company> => {
        const response = await apiClient.post<Company>('/company/create', { name, max_seats: maxSeats });
        return response.data;
    },

    join: async (inviteCode: string): Promise<Company> => {
        const response = await apiClient.post<Company>('/company/join', { invite_code: inviteCode });
        return response.data;
    },

    getSeats: async (): Promise<CompanySeatsResponse> => {
        const response = await apiClient.get<CompanySeatsResponse>('/company/seats');
        return response.data;
    }
};
